pandas/tests/strings/test_find_replace.py

```diff
@@ -148,6 +152,81 @@ def test_contains_na_kwarg_for_nullable_string_dtype(
     tm.assert_series_equal(result, expected)
 
 
+def test_contains_moar(any_string_dtype):
+    # PR #1179
+    s = Series(
+        ["A", "B", "C", "Aaba", "Baca", "", np.nan, "CABA", "dog", "cat"],
+        dtype=any_string_dtype,
+    )
+
+    result = s.str.contains("a")
+    expected_dtype = "object" if any_string_dtype == "object" else "boolean"
+    expected = Series(
+        [False, False, False, True, True, False, np.nan, False, False, True],
+        dtype=expected_dtype,
+    )
+    tm.assert_series_equal(result, expected)
+
+    result = s.str.contains("a", case=False)
+    expected = Series(
+        [True, False, False, True, True, False, np.nan, True, False, True],
+        dtype=expected_dtype,
+    )
+    tm.assert_series_equal(result, expected)
+
+    result = s.str.contains("Aa")
+    expected = Series(
+        [False, False, False, True, False, False, np.nan, False, False, False],
+        dtype=expected_dtype,
+    )
+    tm.assert_series_equal(result, expected)
+
+    result = s.str.contains("ba")
+    expected = Series(
+        [False, False, False, True, False, False, np.nan, False, False, False],
+        dtype=expected_dtype,
+    )
+    tm.assert_series_equal(result, expected)
```

Comment:
could parmaeterize at some point

Reply:
sure. not doing too many changes in one pass to keep the diff simpler